### DISCUSSION TITLE
Is it possible to dump opentelemetry config in the runtime?

### DISCUSSION THREAD

**Discussion by magicliang** [open-telemetry/opentelemetry-java-instrumentation#6284] [Q&A] [answered]
we have a complicated dynamic script to use different configuration in different cases.
Is it possible to dump the configuration opentelemetry is using so that we can debug whether our script work as we expect.

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @magicliang ,
No, currently it is not possible. There is a story #1551 about adding an option to do that (that would most likely be a system prop/env var).

**Reply by magicliang** [reply]
yes，like an jvm option to print all jvm args is all developer need finally as this story describe.

**Reply by magicliang** [reply]
actually what I think better is an api like jmx api so that when all configurations are merged and overriden, we can dump them out to see with multiple config approach what we get finally.

**Reply by trask** [reply]
@magicliang if we dump the final properties (after merging all configs) would that solve the issue? I think JMX would be useful if we supported updating properties, but only for dumping them it seems generally more convenient to set a flag compared to connecting to JMX

**Reply by anuraaga** [reply]
Currently we log the SDK's final configuration, reflecting any SPIs, at debug level at startup. There are some pieces missing like logs config but is this what you are looking for?
